### DISCUSSION TITLE
認証認可のAPIをoauth2対応ライブラリで叩きたい

### DISCUSSION THREAD

**Discussion by nakano16180** [increments/qiita-discussions#170] [要望・アイデア・不具合]
現状のAPIだとリクエスト投げるときのContent-typeや返ってくるレスポンスの形式がoauth2の仕様と違っているためライブラリで叩けないのが少し不便。

**Comment by kskwtnk** [comment]
@nakano16180 
ご意見ありがとうございます！

認証認可のAPIは長い間手を入れられておらず、お手数をおかけします 🙏 
ですが直近でOAuth 2.0に対応できる予定がなく……申し訳ありません！

というのも

- 既存の認証認可APIは残しておく必要がある
- OAuth 2.0に対応する場合新規エンドポイントを作成する
    - 既存のプロジェクトとの兼ね合いやメンテナンス対象の増加を考えると（今でもあまりしっかりできていないため）難しそう

という話になりました

せっかくご意見いただいた中で申し訳ありませんが、ご理解いただけますと幸いです 🙇
